src/irmin/tree.ml

```diff
@@ -924,6 +924,17 @@ module Make (P : Backend.S) = struct
       | Error _ as e -> e
       | Ok m -> Ok (StepMap.bindings m)
 
+    let update_bindings updates bindings =
+      let bindings =
+        Seq.filter (fun (s, _) -> not (StepMap.mem s updates)) bindings
```

Comment:
Let's add a comment that highlights the fact that the `mem` calls may be expensive

Reply:
added a [comment](https://github.com/mirage/irmin/pull/1532/files#diff-b69187f706f079406ffcf31edc21275bdee6ed99b12094a44381876e94eb203aR928), let me know if its ok.

Reply:
Good